### ISSUE TITLE
How to show Accuracy on tensorboard and training log using tensorflow object detection api

### ISSUE THREAD

**Issue by good74152** [tensorflow/models#7456]
### System information
- **What is the top-level directory of the model you are using**:
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow installed from (source or binary)**: pip
- **TensorFlow version (use command below)**: tensorflow-gpu 1.13.1
- **CUDA/cuDNN version**: CUDA 10.0, cuDNN 7.6.2
- **GPU model and memory**: NVIDIA Quadro P1000

I am using faster_rcnn_inception_v2_pets.config with model_main.py , but i have no idea to show the classifier accuracy with test data

Have anyone know how to display the accuracy about test data on training log?

another question is how to display the accuracy about test data on tensorboard like this
![image](https://user-images.githubusercontent.com/39672039/63147724-9ce9a880-c031-11e9-9685-44785510ee1f.png)

thanks

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
Exact command to reproduce

**Comment by good74152** [comment]
@tensorflowbutler 
1. What is the top-level directory of the model you are using:
   (1) C:/tensorflow/models/research/object_detection/training/ (I put config、labelmap、training checkpoint here)
   (2) C:/tensorflow/models/research/object_detection/image/ (I put train、test image folder here)
   (3) C:/tensorflow/models/research/object_detection/ (I put test.record、train.record here)
   (4) C:/tensorflow/models/research/object_detection/faster_rcnn_inception_v2_coco_2018_01_28/ (I put pretrained model here)
2. Have I written custom code (as opposed to using a stock example script provided in TensorFlow): NO

**Comment by benouinirachid** [comment]
Probably, this is the solution of your issue: 
https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/training.html#monitor-training-job-progress-using-tensorboard

**Comment by dilvishjohn** [comment]
This is not a solution tensorboard from box has only loss graphics and no accuracy.
I have same problem. If someone know how to track accuracy during learing help me please.

**Comment by syafiq32** [comment]
i have problem no accuracy show on tensorboard...any idea to fix it...

**Comment by dilvishjohn** [comment]
I have fixed accuracy on tensorflow for object detection api branch r1.13
and tensorflow 1.15.2 and tensorboard 1.16.0
maybe my way help you.

in config  of model
enable some metric I use this by defaul
from this acticle
https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/training.html#evaluating-the-model-optional

eval_config: {
    metrics_set: "coco_detection_metrics"
	num_examples: 9
}

start train here command
python model_main.py -alsologtostderr --model_dir=training\ --pipeline_config_path=training/1.config 

it will be generate need eval folder with evaluation data inside training folder
run this command from this folder model\research\object detection\ 
 tensorboard --logdir=training --host localhost

Profit!!!

**Comment by syafiq32** [comment]
thank you @dilvishjohn

**Comment by syafiq32** [comment]
i done following your step @dilvishjohn  ...... why graph not show like below .....
![Capture](https://user-images.githubusercontent.com/62755514/80707797-23d9a500-8b1d-11ea-9a73-4ff22c1f556a.PNG)

**Comment by dilvishjohn** [comment]
cause it's different metrics...
here supported metrics of object detection.
I think you need found metrics from this graphs or create them by you own, but it's hard and i don't know how it can be done. I sutisfied by coco metrics for me
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/evaluation_protocols.md

**Comment by syafiq32** [comment]
it so okey.... thank you @dilvishjohn for your  time read my problem and give suggention to me.... i will try later....

**Comment by kevinkwabena** [comment]
> i done following your step @dilvishjohn ...... why graph not show like below .....
> ![Capture](https://user-images.githubusercontent.com/62755514/80707797-23d9a500-8b1d-11ea-9a73-4ff22c1f556a.PNG)

is this in object detection

**Comment by anacondabitch** [comment]
or you can review the line 58 in evaluator.py located in models-1.13.0.zip\models-1.13.0\research\object_detection\legacy fodler

**Comment by JackDanHollister** [comment]
I would still be interested in a solution to this if anyone can help with this.

Ive spent the lat couple of days searching for iformation on this subjec and this thread is the closest anyone has come to discussing this!

**Comment by IamExperimenting** [comment]
Hey @dilvishjohn, thanks for your recommendation. Could you please provide the folder structure? because I didn't understand this bit "it will be generate need eval folder with evaluation data inside training folder" which you have mentioned in your previous comment.

**Comment by llingtingzhe** [comment]
does anyone know how to display the accuracy on tensorbord?

**Comment by Annieliaquat** [comment]
plz need answer for this question. does anyone know how to display the accuracy on tensorbord?

**Comment by raulperezalejo** [comment]
Standing here too looking for this answer. There may be a way to see the "history" the same way i do when "fitting" my model manually.
